cql3/Cql.g

```diff
@@ -460,7 +460,9 @@ unaliasedSelector returns [uexpression s]
        | f=functionName args=selectionFunctionArgs { tmp = function_call{std::move(f), std::move(args)}; }
        | K_CAST      '(' arg=unaliasedSelector K_AS t=native_type ')'  { tmp = cast{.style = cast::cast_style::sql, .arg = std::move(arg), .type = std::move(t)}; }
        )
-       ( '.' fi=cident { tmp = field_selection{std::move(tmp), std::move(fi)}; } )*
+       ( '.' fi=cident { tmp = field_selection{std::move(tmp), std::move(fi)}; }
+       | '[' sub=term ']' { tmp = subscript{std::move(tmp), std::move(sub)}; }
```

Comment:
This is a little ambitious as it brings in terms that were only used in WHERE/IF into selectors.

Overall it's a good direction but it needs significant testing.

One thing to watch out for is column[udf(?]], since udf evaluation requires with_thread_if_needed(). It may be already provided since we support udf selectors, but the if_needed part needs to be tested.

Also exclude aggregate functions.

Reply:
We also need to test what happens when the user attempts to subscript something which isn't a map or set - such as a list (which Cassandra doesn't support subscripting!) or something silly like an integer. Also worth checking what happens when the value or the subscript is a null (I don't remember if Cassandra's unit tests had all these things, most likely it did).

Reply:
I don't think most integers are silly, but otherwise agree.

Reply:
I added a test verifying that `SELECT m[udf]` works.
also cases of a subscript on a non-map (or list) column which fails as expected.
a null subscript behaves differently than cassandra - we return null but cassandra returns an error

Reply:
Yes, that's documented (docs/cql/cql-extensions.md).